### ISSUE TITLE
TypeError with unflatten when destination property is null

### ISSUE THREAD

**Issue by A** [hughsk/flat#25]
```js
// ok
flat.unflatten({ 'a.b': 'c' }); // { a: { b: 'c' } }
// not ok
flat.unflatten({ a: null, 'a.b': 'c' }); // TypeError: Cannot set property 'b' of null
```

Probably, it happens because of `'object' === typeof null`, but I didn't see the source

**Comment by timoxley** [comment]
Released in 2.0.0. I've bumped major on this because the behaviour around overwriting non-object values is changed from 1.x.

**Comment by A** [comment]
Thank you)
